src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2396,21 +2396,25 @@ private void visitProperties(
      * property in the loop above, and must guess on the actual return type of the iterator method.
      *
      * <p>TODO(rado): This code will almost never work with iclutz, because IterableIterator and the
-     * class that implements it would be part of different complitation. Investigate whether this
-     * can be syntactically resolved.
+     * class that implements it would be part of different compilation. Investigate whether this can
+     * be syntactically resolved.
      */
     private void maybeEmitSymbolIterator(JSType instanceType) {
       if (instanceType == null) {
         return;
       }
-      // iteratorIterableType and iterableType can be null if they were not found in the current
-      // run's externs definitions.
+      // iteratorIterableType will be null if not found in the current run's externs definitions.
       JSType implemented;
       String returnType;
       if (iteratorIterableType != null && instanceType.isSubtype(iteratorIterableType)) {
         implemented = iteratorIterableType;
         returnType = "IterableIterator";
-      } else if (iterableType != null && instanceType.isSubtype(iterableType)) {
+        // It appears that iterableType is always defined. However, a simple extending unknown
+        // record is reported as subtype of iterable for unknown reasons, so in partial mode we
+        // do not emit the iterator signature.
+      } else if (iterableType != null
```

Comment:
The type doesn't have any information that it's a record?

Reply:
It does. Are you suggesting narrowing the exception to record types only? I guess I can try that tomorrow.

Reply:
Spend some time looking into better checks specifically for interface/record, but for some unknown reason `instanceType.isInterface()` returns `false`. Added some comments and tests for interfaces too, but the internal logic is the same.